test/testdrive/protobuf-evolution.td

```diff
@@ -0,0 +1,127 @@
+# Copyright Materialize, Inc. and contributors. All rights reserved.
+#
+# Use of this software is governed by the Business Source License
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+# Test that Protobuf message evolution is handled sensibly.
+
+# Ingest one message with a very simple schema.
+
+$ set schema
+syntax = "proto3";
+
+message Message {
+    int32 i = 1;
+}
+
+$ file-append path=evolution.proto
+\${schema}
+
+$ protobuf-compile-descriptors inputs=evolution.proto output=evolution.pb
+
+$ schema-registry-publish subject=testdrive-evolution-${testdrive.seed}-value schema-type=protobuf
+\${schema}
+
+$ kafka-create-topic topic=evolution partitions=1
+
+> CREATE MATERIALIZED SOURCE evolution FROM
+  KAFKA BROKER '${testdrive.kafka-addr}' TOPIC 'testdrive-evolution-${testdrive.seed}'
+  FORMAT PROTOBUF USING CONFLUENT SCHEMA REGISTRY '${testdrive.schema-registry-url}'
+
+$ kafka-ingest topic=evolution format=protobuf descriptor-file=evolution.pb message=.Message confluent-wire-format=true
+{"i": 1}
+
+> SELECT * FROM evolution
+i  mz_offset
+------------
+1  1
+
+# Adding new fields and renaming existing fields is safe. New fields won't be
+# reflected in the source, though.
+
+$ set schema
+syntax = "proto3";
+
+message Message {
+    int32 i_renamed = 1;
+    bool extra = 2;
+}
+
+$ file-delete path=evolution.proto
+$ file-append path=evolution.proto
+\${schema}
+
+$ protobuf-compile-descriptors inputs=evolution.proto output=evolution.pb
+
+$ schema-registry-publish subject=testdrive-evolution-${testdrive.seed}-value schema-type=protobuf
+\${schema}
```

Comment:
probably want a comment here that says this is where the evolution happens

Reply:
I think I may have rewritten this since you looked at it; turns out the Confluent Schema Registry in mzcompose is much stricter about compatibility than the one I had running locally.